lib/spack/spack/relocate.py

```diff
@@ -0,0 +1,319 @@
+        nrpaths.append(nrpath)
+    for dep in deps:
+        ndep = dep.replace(old_dir, new_dir)
+        ndeps.append(ndep)
+    return nrpaths, ndeps, id
+
+
+def modify_macho_object(path_name, old_dir, new_dir, relative):
+    """
+    Modify MachO binaries by changing rpaths,and id and dependency lib paths.
+    Examines the output of otool -l for these three patterns
+    cmd LC_ID_DYLIB
+    cmdsize 160
+    name /Users/gartung/spack-macdev/opt/spack/darwin-x86_64/clang-7.0.2
+    -apple/tcl-8.6.5-xfeydlhaojmei6iws2rnxndvriym242k/lib/libtcl8.6.dylib
+    (offset 24)
+    cmd LC_LOAD_DYLIB
+    cmdsize 160
+    name /Users/gartung/spack-macdev/opt/spack/darwin-x86_64/clang-7.0.2
+    -apple/zlib-1.2.8-cyvcqvrzlgurne424y55hxvfucvz2354/lib/libz.1.dylib
+    (offset 24)
+    cmd LC_RPATH
+    cmdsize 128
+    path /Users/gartung/spack-macdev/opt/spack/darwin-x86_64/clang-7.0.2
+    -apple/xz-5.2.2-d4ecxpuzf2g3ycz3cnj3xmdj7zdnuqwb/lib
+    (offset 12)
+    the old install dir in LC_LOAD_DYLIB is replaced with the new install dir
+    using install_name_tool -id newid binary
+    the old install dir in LC_LOAD_DYLIB is replaced with the new install dir
+    using install_name_tool -change old new binary
+    the old install dir in LC_RPATH is replaced with the new install dir using
+    install_name_tool  -rpath old new binary
+    """
+    if 'libgcc_' in path_name:
+        return
+    rpaths, deps, idpath = macho_get_paths(path_name)
+    id = None
+    nrpaths = []
+    ndeps = []
+    if relative:
+        nrpaths, ndeps, id = macho_make_paths_rel(path_name,
+                                                  old_dir, rpaths,
+                                                  deps, idpath)
+    else:
+        nrpaths, ndeps, id = macho_replace_paths(old_dir, new_dir, rpaths,
+                                                 deps, idpath)
+    st = os.stat(path_name)
+    wmode = os.access(path_name, os.W_OK)
+    if not wmode:
+        try:
+            os.chmod(path_name, st.st_mode | stat.S_IWUSR)
+        except OSError as e:
+            tty.warn(e)
+    install_name_tool = which('install_name_tool')
+    if id:
+        try:
+            install_name_tool('-id', id, path_name, output=str, err=str)
+        except ProcessError as e:
+            tty.warn(e)
+
+    for orig, new in zip(deps, ndeps):
+        try:
+            install_name_tool('-change', orig, new, path_name)
+        except ProcessError as e:
+            tty.warn(e)
+
+    for orig, new in zip(rpaths, nrpaths):
+        try:
+            install_name_tool('-rpath', orig, new, path_name)
+        except ProcessError as e:
+            tty.warn(e)
+    try:
+        os.chmod(path_name, st.st_mode)
+    except OSError as e:
+        tty.warn(e)
+    return
+
+
+def get_filetype(path_name):
+    """
+    Check the output of the file command for given string.
+    """
+    bash = which('bash')
+    output = bash('-c', 'LC_ALL=C && file -b -h %s' % path_name,
```

Comment:
It produces a warning and returns type None. This means that nothing would be listed as needing relocation.